src/react.ts

```diff
@@ -34,6 +34,10 @@ const useAffectedDebugValue = (
   useDebugValue(pathList.current)
 }
 
+// This is required only for performance.
+// Ref: https://github.com/pmndrs/valtio/issues/519
+const targetCache = new WeakMap()
```

Comment:
Just for my understanding, doesn't the `configurable: true` fix to `createSnapshot` mean that snapshots shouldn't be getting copied anymore, hence the `targetCache` shouldn't be necessary?

Or are you thinking there could be non-snapshot objects that we come across that might be copied and/or lets keep it for good measure just given how important avoiding the copies is?

Reply:
I hoped so, but `needsToCopyTarget` function is very slow for #519 case, compared to `Object.isFrozen`.

Reply:
Oh jeez, I see, even merely evaluating "do I need to copy?" is itself expensive... :-| 

That makes sense...